### DISCUSSION TITLE
Audio

### DISCUSSION THREAD

**Discussion by cukal** [tesla-android/issue-tracker#57] [Q&A] [answered]
I installed 2022.25.1, following the guide, pretty much a fresh install, all good. I'm missing something on how to get the audio. Is it supposed to just work through the browser? I'm confused as the installation guide mentions audio will go through the Tesla browser but Tesla browser does not allow any audio at all unless a player goes fullscreen and the car is in park. Perhaps it is a regional thing / limitation? 

Any way, great work!

**Comment by galosre** [comment]
As for me the audio did not work also at the beginning.
 I did again the installation again from both Linux and android and now everything working navigation audio phone call  Dsub app for my music server (working from all over the world) even phone call, and all of  this from android auto I do not use apple
so basically you connect the wifi to android auto and also connect the CarlinKit CPC200-Autokit, CPC200-CCPA the Bluetooth need to be connected to your car as usual and that it's so far working perfect.
 I did look at the power issue yet, but now I am happy it is working finally.

**Reply by cukal** [reply]
Do I need to connect both the car _and_ my phone to the carlink bluetooth?

**Reply by galosre** [reply]
Asuuming you have your phone connected as usual to your car if you got the wire connection, you do nothing maybe click on the middle icon from autobox  to initiate the connection for the wireless to I go to Bluetooth on my phone and click Autobox-xxxx should get you going but start with wire first.

**Reply by cukal** [reply]
Thx. The music apps all work (play audio through car) when doing that, but no audio output from waze or google maps. It seems the audio players allow selecting a different audio output even when the phone is in carplay mode but waze/google maps default to carplay when it detects carplay is being used, any other option is greyed out. I guess if waze allowed to select an audio output that would work.

**Comment by galosre** [comment]
Edit
I noticed the audio is someway better if you use the CPC200-CCPA wired and not wireless for my case music steaming maybe a Bluetooth bandwidth limit(sound stop every 2second for a  third of second on the CPC200 wireless)

**Comment by rrlevy** [comment]
I'm having a lot of trouble with the audio! After leaving the car and letting the system turned on (car not sleeping), when I get back I need to relaunch the Audio Capture app. Sometimes it also crashes by itself during drive, while using CarPlay

**Reply by mikegapinski** [reply]
Will take a look into this, maybe there is something wrong with how the websocket connection is handling new clients on reload

**Comment by cukal** [comment]
Car updated to 2022.16.3 from 2022.12.3.2 and lo & behold, when driving this morning Waze is _suddenly_ outputting audio to car bluetooth. Phone wasn't connected to autolink bluetooth, normally it locks on to that but not this morning. I have no idea why it now suddenly works...

**Reply by galosre** [reply]
Probably just Waze running in your phone so this is normal you get the audio tru Bluetooth

**Comment by cukal** [comment, marked as answer]
I've been using it since the first release. Previously to start it up: my phone connected over bluetooth to Autolink & Model Y, and over Wifi to Autolink, both automatically (after setting it up). No audio through car, disconnecting bluetooth to Autolink just stopped CarPlay all together. After the update to `2022.16.3` I just got in the car, did the same thing to start it up: started Autolink app, Wifi icon turns green, tapped the Autolink bluetooth icon to have it detect the phone and Autolink CarPlay starts up after it detects the phone. Only difference is now my phone shows it's no longer connected to Autolink over bluetooth but it's only connected to Model Y over bluetooth and to Autolink over wifi and there you have it, Waze output to Model Y speakers. I'm over the moon, this car is _finally_ complete!

**Comment by galosre** [comment]
As for me the audio is also resolved but the main issue is to be able to connect the cp200 wirelessly most of the time got the spinning wheel in the cp200, so my only option is to connect the cable from the car link to the phone so once it is connected with wire,  I wait 5 mn and remove the cable and then it stay connected wirelessly.
So if you can describe your method, it would be very helpful.

**Reply by cukal** [reply]
Only wifi & CarPlay never succeeded until I paired the phone also over bluetooth to autolink. Now with the audio phone connects over wifi, the autolink wifi icon is green but I also need to enable bluetooth on my phone otherwise it just spins & spins, after enabling bluetooth on my phone I tap on the autolink bluetooth icon and it spins for a few seconds and starts CarPlay. Phone is no longer connected to autolink bluetooth (it used to) but it's now connected to Tesla bluetooth. Seems to be automagically now. I'm crossing fingers it won't stop working and revert back to no audio. If I could disable updates I would ;)

**Reply by mikegapinski** [reply]
The fastest way to enable CarPlay for me is either to use the auto connect settings in auto kit - it works for most of the time. When it does not you can go to Bluetooth settings on your iPhone and tap on the AutoKit device, it will connect instantly and CarPlay will launch. 

Tapping the bluetooth icon on the Android app doesn't actually do anything, when wifi connects before Bluetooth it just waits for an iOS device to pair automatically, it takes some time if Tesla connected to your iPhone for sound ahead of that.

AutoKit software updates should make the experience more OEM feeling at some point, they've improved a lot since December

**Reply by cukal** [reply]
As of now, it pretty much feels OEM to me 😃, most certainly since the 2nd release. I used to not enable auto-connect on the first release as it did involve going into parking, trying & re-trying and sometimes rebooting the android pi but now it's just flawless each-and-every-time I get in the car, I don't even wait for it to connect anymore, just start driving & refresh the page while I'm already driving hence I'm going to enable auto-connect again. 

Thanks again for creating this Michal, it was a nice car before, but now it's the perfect car!

**Reply by galosre** [reply]
Today for me car link was total garbage a lot of trouble to have it connected even with the cable, 
So how does it connect Thu wifi,
 From where does it get the PW is it tru Bluetooth?

**Comment by galosre** [comment]
Since the cp200 did not connect wirelessly anymore I unpaired it and pair it again now my experience is similar to Michał Gapiński so much better Thank you
